### PR TITLE
Set close-on-exec flag to prevent fd leakage for command execution

### PR DESCRIPTION
**Pull request by kubo39**

### CODE REVIEW

File: source/eventcore/drivers/posix/sockets.d [hunk 1]
```diff
@@ -927,6 +930,13 @@ private void setSocketNonBlocking(SocketFD sockfd)
 	}
 }
 
+private void setSocketCloseOnExec(SocketFD sockfd)
+@nogc nothrow {
+	version (Windows) {} else {
+		() @trusted { fcntl(cast(int)sockfd, F_SETFL, O_CLOEXEC, 1); } ();
```
Comment (kubo39) [inline source/eventcore/drivers/posix/sockets.d]: Ooops...sorry, O_CLOEXEC is defined only FreeBSD in druntime.

Comment (kubo39) [reply source/eventcore/drivers/posix/sockets.d]: fixed https://github.com/vibe-d/eventcore/pull/38/commits/2f3f7b1773f88539d0dc3ed8dce1fcecaa950d27

**Comment by s-ludwig** [comment]
Thanks! The changes look good and this should indeed be the default. The question is whether the OSX failures are just flaky tests or are somehow caused by the changes. I had restarted them once, but they failed again in the same way. I'll try once more.

**Comment by kubo39** [comment]
Hmm, it is strange. Although build failed on #38 in the same reason, it is a change not related to OSX.

**Comment by s-ludwig** [comment]
The failures also occur on commits that succeeded before, so I'll just merge #39 forcefully. The only difference appears to be the OSX/select build, which has hung consistently in the 0-tcp test for this PR so far. Once master is green again I can investigate that further.

**Comment by s-ludwig** [comment]
Ah, got it! The problem is that `setSocketCloseOnExec` removes the `O_NONBLOCK` flag, so the code needs to be changed to set both flags at once. The hang is caused by the order of write/read sometimes being reversed.

The deception happens because of the `1` argument to `fnctl`, which is actually superfluous - the third argument is actually a bit mask and there is no fourth argument.

I've created a new branch that addes the required fixes: https://github.com/vibe-d/eventcore/commits/cloexec

**Pull request closed without merging**